pkg/storage/batch.go

```diff
@@ -568,6 +570,13 @@ func (it *sampleBatchIterator) buildIterators(chks map[model.Fingerprint][][]*La
 	return result, nil
 }
 
+func filteredLabels(ls labels.Labels) labels.Labels {
+	return labels.NewBuilder(ls).
+		Del(labels.MetricName).
+		Del(distributor.ShardLbName).
```

Comment:
Question because I never used `Del` myself: if `stream_shard` isn't enabled the label won't be present anyway; will that be a problem with Del? Or it just ignores it?

Reply:
Del doesn't do anything if the label doesn't exist.